mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -1818,6 +1817,78 @@ class LowerEqualRewrite : public OpRewritePattern<graphblas::EqualOp> {
   };
 };
 
+class LowerVectorArgMinMaxOpRewrite : public OpRewritePattern<graphblas::VectorArgMinMaxOp> {
+public:
+  using OpRewritePattern<graphblas::VectorArgMinMaxOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(graphblas::VectorArgMinMaxOp op, PatternRewriter &rewriter) const override {
+    // TODO we get seg faults if given a size 0 vector or a sparse vector with no non-zero values.
+    Location loc = op->getLoc();
+
+    Value input = op.vec();
+    RankedTensorType inputType = input.getType().dyn_cast<RankedTensorType>();
+    ArrayRef<int64_t> inputShape = inputType.getShape();
+    
+    Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
+    Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
+    Type indexType = rewriter.getIndexType();
+    Type int64Type = rewriter.getIntegerType(64);
+    Type memref1DI64Type = MemRefType::get({-1}, int64Type);
+    int64_t inputLength = inputShape[0];
+    Value endPosition;
+    if (inputLength == -1) {
```

Comment:
Ideally, `graphblas.size` would handle this optimization.

Reply:
Upon further investigation, `memref.dim` already handles this optimization, so `graphblas.size` doesn't actually need to handle this. Thus, I should remove this optimization here.

Reply:
Good find!

Reply:
https://github.com/metagraph-dev/mlir-graphblas/pull/111/commits/2a90f1977b878dfa50d390c629de204ba929268f addresses this issue.